File: my-solutions/0153-find-minimum-in-rotated-sorted-array/solution.py
class Solution:
    def findMin(self, nums: List[int]) -> int:

        l, r  = 0, len(nums) - 1
        size = r
        
        while(l <= r):
            if l == r:
                return nums[l]
            if nums[l] < nums[r]:
                return nums[l]
            mid = int((l + r)  / 2)
            if mid != 0 and mid != size and nums[mid] < nums[mid - 1] and nums[mid] < nums[mid + 1]:
                return nums[mid]
            elif mid != 0 and mid != size and nums[mid] > nums[mid - 1] and nums[mid] > nums[mid + 1]:
                return nums[mid + 1]
            elif nums[mid] < nums[l]:
                r = mid - 1
            else:
                l = mid + 1

        return 1000
        

File: my-solutions/0153-find-minimum-in-rotated-sorted-array/solution.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int l = 0, r = size(nums) - 1 - 1;
        int rotate = -1;
        while(l <= r) {
            int mid = (l + r) / 2;
            if(nums[mid] > nums[mid + 1]) {
                rotate = mid;
                break;
            } else if(nums[mid] < nums[0]) {
                r = mid - 1;
            } else {
                l = mid + 1;
            }
        }


        if(rotate == -1) {
            return nums[0];
        } else {
            return nums[rotate + 1];
        }
    }
};
